pyiron_atomistics/vasp/vasprun.py

```diff
@@ -538,18 +537,108 @@ def parse_structure_to_dict(self, node, d):
 
             if leaf.tag == "varray" and leaf.attrib["name"] == "selective":
                 d["selective_dynamics"] = self._parse_2d_matrix(leaf, vec_type=bool)
+    
+    def _parse_INCAR_generator_parameters(self, val_type, val):
+        """
+        Helper function to convert a Vasprun parameter into the proper type.
+        Boolean, int and float types are converted.
+
+        Args:
+            val_type: Value type parsed from vasprun.xml.
+            val: Actual string value parsed for vasprun.xml.
+        """
+        if val_type == "logical":
+            return val == "T"
+        if val_type == "int":
+            return int(val)
+        if val_type == "string":
+            return val.strip()
+        return float(val)
+
 
+    def _parse_INCAR_generator_vector_parameters(self, val_type, val, filename, param_name):
+        """
+        Helper function to convert a Vasprun array-type parameter into the proper
+        type. Boolean, int and float types are converted.
+
+        Args:
+            val_type: Value type parsed from vasprun.xml.
+            val: Actual string value parsed for vasprun.xml.
+            filename: Fullpath of vasprun.xml. Used for robust error handling.
+                E.g., if vasprun.xml contains *** for some Incar parameters,
+                the code will try to read from an INCAR file present in the same
+                directory.
+            param_name: Name of parameter.
+
+        Returns:
+            Parsed value.
+        """
+        err = ValueError("Error in parsing vasprun.xml")
+        if val_type == "logical":
+            val = [i == "T" for i in val.split()]
+        elif val_type == "int":
+            try:
+                val = [int(i) for i in val.split()]
+            except ValueError:
+                # Fix for stupid error in vasprun sometimes which displays
+                # LDAUL/J as 2****
+                val = self._parse_from_incar(filename, param_name)
+                if val is None:
+                    raise err
```

Comment:
```suggestion
                    raise err from None
```
This will clear up the stacktrace.

It feels a bit strange, to pre define the exception.  It works and there's no down side, though.